src/adapter/src/coord.rs

```diff
@@ -2942,6 +2952,14 @@ impl Coordinator {
             let idle_metric = self.metrics.queue_busy_seconds.with_label_values(&[]);
             let last_message_watchdog = Arc::clone(&last_message);
 
+            // TODO(jkosh44) Do this in the background.
+            if let Some(retention_period) = storage_usage_retention_period {
+                let prune_start = Instant::now();
+                info!("startup: coord serve: storage usage prune beginning");
+                self.prune_storage_usage_events_on_startup(retention_period).await;
+                info!("startup: coord serve: storage usage prune complete in {:?}", prune_start.elapsed());
```

Comment:
Annoyingly, this work happens on a completely different thread than all the other startup work, so it is not included in any of the other timings. Instead you have to add it manually to those timings.